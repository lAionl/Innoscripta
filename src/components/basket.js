import React, { Component } from "react";


export class Basket extends Component {

    render() {
        const { basket, deletePizza } = this.props;
        const totalPrice = basket.reduce((accumulator, elem) => accumulator + elem.price, 0);
        return (
            <>
                <div className="alert alert-success mb-2 mt-2 ml-3 mr-3">In your basket:</div>
                <ul className="list-group list-group-flush mb-3 mt-3 ml-3 mr-3">
                    {basket.map((elem, i) =>
                        <li className="list-group-item mb-1 mt-1 ml-1 mr-1" key={i}>
                            {`${elem.name} - ${elem.price} $`}
                            <button 
                                className="btn btn-danger btn-sm mb-1 mt-1 ml-1 mr-1"
                                onClick={() => deletePizza(elem)}
                            >{' x '}</button>
                        </li>
                    )}
                </ul>
                <div className="alert alert-primary mb-3 mt-3 ml-3 mr-3">
                    <p>{`Total for ${basket.length} ${basket.length === 1 ? 'pizza' : 'pizzas'}: ${totalPrice}$`}</p>
                    <p>{`With delivery: ${totalPrice + 3}$`}</p>
                </div>
            </>
        );
    }
}
