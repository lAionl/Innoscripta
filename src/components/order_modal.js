import React, { Component } from "react";
import ReactDOM from 'react-dom';

export class OrderModal extends Component {
    constructor(props) {
        super(props);
        this.el = document.createElement('div');
        this.state = {
            address: '',
            phone: '',
            name: '',
            showSuccess: false,
        };
        this.getCountPizzas = this.getCountPizzas.bind(this);
    }
    componentDidMount() {
        const modalRoot = document.getElementById('modal-root');
        modalRoot.appendChild(this.el);
    }

    componentWillUnmount() {
        const modalRoot = document.getElementById('modal-root');
        modalRoot.removeChild(this.el);
    }

    getCountPizzas(basket) {
        const pizzasInBasket = basket.map(item => item.name).reduce((acc, el) => {
            acc[el] = (acc[el] || 0) + 1;
            return acc;
        }, {});
        let text = [];
        for (let [key, value] of Object.entries(pizzasInBasket)) {
            text = [...text, `${key}: ${value}, `];
        }
        return text;
    }

    render() {
        const { onClose, basket } = this.props;
        const { address, phone, name, showSuccess } = this.state;
        const countPizzas = this.getCountPizzas(basket);
        return (
            <>
            {ReactDOM.createPortal(
                (<div className="modal-dialog modal-dialog-centered">
                    <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title">{'Order'}</h5>
                            <button onClick={onClose} className="close">
                                <span>{' X '}</span>
                            </button>
                        </div>
                        {!showSuccess && (
                            <div className="modal-body">
                                <p>{`In your basket ${basket.length} ${basket.length === 1 ? 'pizza.': 'pizzas.'}`}</p>
                                <p>{`${basket.length === 1 ? 'There is:' : 'There are:'} ${countPizzas}`}</p>
                                <div className="form-group">
                                    <label htmlFor="name">{'Name'}</label>
                                    <input
                                        id="name"
                                        className="form-control"
                                        value={name}
                                        onChange={val => this.setState({ name: val.target.value })}
                                    />
                                </div>
                                <div className="form-group">
                                    <label>{'Address'}</label>
                                    <textarea
                                        className="form-control"
                                        value={address}
                                        onChange={val => this.setState({ address: val.target.value })}
                                    />
                                </div>
                                <div className="form-group">
                                    <label>{'Phone number'}</label>
                                    <input
                                        value={phone}
                                        onChange={val => this.setState({ phone: val.target.value })}
                                        type="tel"
                                        className="form-control"
                                        pattern="[0-9]{3}-[0-9]{2}-[0-9]{3}"
                                    />
                                </div>
                            </div>
                        )}
                        {showSuccess && (
                            <div className="modal-body alert-success">{'Your order has been sent! Thank you!'}</div>
                        )}
                        <div className="modal-footer">
                            <button onClick={onClose} className="btn btn-secondary">{'Close'}</button>
                            {!showSuccess && (
                                <button
                                    onClick={() => this.setState({ showSuccess: true })}
                                    disabled={!name || !address || !phone}
                                    className="btn btn-primary">{'Send an order'}
                                </button>
                            )}
                        </div>
                    </div>
                </div>), this.el
            )}
            </>
        );
    }
}
