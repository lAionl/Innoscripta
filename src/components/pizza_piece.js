import React, { Component } from "react";


export class PizzaPiece extends Component {

    render() {
        const { pizza, addPizza } = this.props;
        return (
            <div className="card border-primary mb-3 mt-3 ml-3 mr-3">
                <img 
                    className="card-img-top"
                    src={pizza.url}
                    style={{ width: '180px', height: '160px'}}
                />
                <div className="card-body">
                    <h5 className="card-title">{pizza.name}</h5>
                    <p className="card-text">{`${pizza.price} $`}</p>
                    <p className="card-text">{pizza.composition}</p>
                    <bitton className="btn btn-primary" onClick={addPizza}>Buy</bitton>
                </div>
            </div>
        );
    }
}
