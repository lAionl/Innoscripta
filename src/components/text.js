import React, { Component } from "react";

export class TextComponent extends Component {

    render() {
        const { tag, content, styleClass } = this.props;
        const TagName = tag;
        return (
            <TagName className={styleClass}>{content}</TagName>
        );
    }
}
