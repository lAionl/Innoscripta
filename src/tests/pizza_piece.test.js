import React from 'react';
import renderer from 'react-test-renderer';
import { PizzaPiece } from '../components/pizza_piece';

test('PizzaPiece', () => {
    const tree = renderer
        .create(
            <PizzaPiece
                pizza={{
                    url: 'http://www.ww.ww',
                    name: 'Mamma Mia',
                    price: 2,
                    composition: 'Some text about composition'
                }}
            />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});