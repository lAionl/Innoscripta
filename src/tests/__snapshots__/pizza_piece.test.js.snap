// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaPiece 1`] = `
<div
  className="card border-primary mb-3 mt-3 ml-3 mr-3"
>
  <img
    className="card-img-top"
    src="http://www.ww.ww"
    style={
      Object {
        "height": "160px",
        "width": "180px",
      }
    }
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Mamma Mia
    </h5>
    <p
      className="card-text"
    >
      2 $
    </p>
    <p
      className="card-text"
    >
      Some text about composition
    </p>
    <bitton
      className="btn btn-primary"
    >
      Buy
    </bitton>
  </div>
</div>
`;
